File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/LoadMoreGroupItemFactory.java
package me.xiaopan.assemblyadaptersample.itemfactory;

import android.view.View;
import android.view.ViewGroup;

import me.xiaopan.assemblyadapter.AssemblyLoadMoreGroupItemFactory;
import me.xiaopan.assemblyadapter.OnGroupLoadMoreListener;
import me.xiaopan.assemblyadaptersample.R;

public class LoadMoreGroupItemFactory extends AssemblyLoadMoreGroupItemFactory {

    public LoadMoreGroupItemFactory(OnGroupLoadMoreListener eventListener) {
        super(eventListener);
    }

    @Override
    public AssemblyLoadMoreGroupItem createAssemblyItem(ViewGroup parent) {
        return new LoadMoreGroupItem(R.layout.list_item_load_more, parent);
    }

    public class LoadMoreGroupItem extends AssemblyLoadMoreGroupItem {
        private View loadingView;
        private View errorView;
        private View endView;

        public LoadMoreGroupItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onFindViews() {
            loadingView = findViewById(R.id.text_loadMoreListItem_loading);
            errorView = findViewById(R.id.text_loadMoreListItem_error);
            endView = findViewById(R.id.text_loadMoreListItem_end);
        }

        @Override
        public View getErrorRetryView() {
            return errorView;
        }

        @Override
        public void showLoading() {
            loadingView.setVisibility(View.VISIBLE);
            errorView.setVisibility(View.INVISIBLE);
            endView.setVisibility(View.INVISIBLE);
        }

        @Override
        public void showErrorRetry() {
            loadingView.setVisibility(View.INVISIBLE);
            errorView.setVisibility(View.VISIBLE);
            endView.setVisibility(View.INVISIBLE);
        }

        @Override
        public void showEnd() {
            loadingView.setVisibility(View.INVISIBLE);
            errorView.setVisibility(View.INVISIBLE);
            endView.setVisibility(View.VISIBLE);
        }
    }
}

File: sample/src/main/java/me/panpf/assemblyadapter/sample/itemfactory/LoadMoreGroupItemFactory.kt
package me.panpf.assemblyadapter.sample.itemfactory

import android.view.View
import android.view.ViewGroup

import me.panpf.assemblyadapter.AssemblyLoadMoreGroupItemFactory
import me.panpf.assemblyadapter.OnGroupLoadMoreListener
import me.panpf.assemblyadapter.sample.R
import me.panpf.assemblyadapter.sample.bindView

class LoadMoreGroupItemFactory(eventListener: OnGroupLoadMoreListener) : AssemblyLoadMoreGroupItemFactory(eventListener) {

    override fun createAssemblyItem(parent: ViewGroup): AssemblyLoadMoreGroupItemFactory.AssemblyLoadMoreGroupItem<*> {
        return LoadMoreGroupItem(R.layout.list_item_load_more, parent)
    }

    inner class LoadMoreGroupItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyLoadMoreGroupItemFactory.AssemblyLoadMoreGroupItem<Int>(itemLayoutId, parent) {
        val loadingView: View by bindView(R.id.text_loadMoreListItem_loading)
        val errorView: View by bindView(R.id.text_loadMoreListItem_error)
        val endView: View by bindView(R.id.text_loadMoreListItem_end)

        override fun getErrorRetryView(): View {
            return errorView
        }

        override fun showLoading() {
            loadingView.visibility = View.VISIBLE
            errorView.visibility = View.INVISIBLE
            endView.visibility = View.INVISIBLE
        }

        override fun showErrorRetry() {
            loadingView.visibility = View.INVISIBLE
            errorView.visibility = View.VISIBLE
            endView.visibility = View.INVISIBLE
        }

        override fun showEnd() {
            loadingView.visibility = View.INVISIBLE
            errorView.visibility = View.INVISIBLE
            endView.visibility = View.VISIBLE
        }
    }
}
